net/socks5/response.go

```diff
@@ -0,0 +1,79 @@
+package socks5
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+)
+
+const MaxResponsePacketSize int = 261
+
+// SOCKS5Reply are the bytes sent in SOCKS5 packets
+// that represent replies from the server to a client
+// request.
+type SOCKS5Reply byte
+
+const (
+	Success                 SOCKS5Reply = 0
+	GeneralFailure          SOCKS5Reply = 1
+	ConnectionNotAllowed    SOCKS5Reply = 2
+	NetworkUnreachable      SOCKS5Reply = 3
+	HostUnreachable         SOCKS5Reply = 4
+	ConnectionRefused       SOCKS5Reply = 5
+	TTLExpired              SOCKS5Reply = 6
+	CommandNotSupported     SOCKS5Reply = 7
+	AddressTypeNotSupported SOCKS5Reply = 8
+)
+
+// Response contains the contents of
+// a response packet sent from the proxy
+// to the client.
+type Response struct {
+	reply    SOCKS5Reply
+	addrType Addr
+	bindAddr string
+	bindPort uint16
+}
+
+// PacketFromResponse converts a SOCKS5Response struct into
+// a packet. If res.reply == Success, it may throw an error on
+// receiving an invalid bind address. Otherwise, it will not throw.
+func PacketFromResponse(res *Response) ([]byte, error) {
```

Comment:
This should get worked into fuzzing--round tripping serialization/deserialization is a great way to find bugs.